sway-lsp/src/core/session.rs

```diff
@@ -332,20 +336,31 @@ impl Session {
     }
 
     /// Create runnables if the `TyProgramKind` of the `TyProgram` is a script.
-    fn create_runnables(&self, typed_program: &ty::TyProgram, decl_engine: &DeclEngine) {
+    fn create_runnables(
+        &self,
+        typed_program: &ty::TyProgram,
+        decl_engine: &DeclEngine,
+        source_engine: &SourceEngine,
+    ) {
         // Insert runnable test functions.
         for (decl, _) in typed_program.test_fns(decl_engine) {
             // Get the span of the first attribute if it exists, otherwise use the span of the function name.
             let span = decl
                 .attributes
                 .first()
                 .map_or_else(|| decl.name.span(), |(_, attr)| attr.span.clone());
-            let runnable = Box::new(RunnableTestFn {
-                span,
-                tree_type: typed_program.kind.tree_type(),
-                test_name: Some(decl.name.to_string()),
-            });
-            self.runnables.insert(runnable.span().clone(), runnable);
+            if let Some(source_id) = span.source_id() {
+                let path = source_engine.get_path(source_id);
```

Comment:
Now that #4961 is merged, we should have an `Option<PathBuf>` to use in the `TokenIdent`. This also means we don't need to pass the source engine into this function now.

Reply:
Hmm, I'm not sure how to get the TokenIdent or even the file path from this type without using the source engine. Do you have any idea?

Reply:
nevermind you're correct, we still need the source engine up until the point of creating the type.